### ISSUE TITLE
Wrong function overload selected when used with array destructuring 

### ISSUE THREAD

**Issue by sooth-sayer** [microsoft/TypeScript#36738]
**TypeScript Version:**  3.7.5

**Search Terms:**
overload destructuring 

**Code**

```ts
interface ITest {
  id: number,
  name: string,
}

const list1: ITest[] = [{ id: 1, name: '1' }, { id: 2, name: '2' }];
const list2: ITest[] = [{ id: 3, name: '3' }];

export function pluck<K extends keyof T, T>(p: K, list: readonly T[]): Array<T[K]>;
export function pluck<P extends string>(p: P): <T>(list: Array<Record<P, T>>) => T[];
export function pluck(...args: any[]): any {
  return [];
}

function test<T>(list1: T[], list2: T[]): T[] {
  return [];
}

const res = [...pluck('id', list2)];
const g = test([1, 2], [...res]);
const g1 = test([1, 2], [...pluck('id', list2)]);
```

**Expected behavior:**
No type errors
**Actual behavior:**
Type 'never' must have a '[Symbol.iterator]()' method that returns an iterator.
on last string

**Playground Link:** http://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=21&pc=50#code/JYOwLgpgTgZghgYwgAgJIBUIGczIN4BQyywAJgFzIgCuAtgEbQA0RVctElOUoA5iwF8CBBAHsQOZABtgOAIyUM2MAG0AusgC8yFXhIVkcpmw6UA5HLPIBxvWUoAmYyHadkZh1YFqA3CPGSMjgOipg46lo6dgYAzM6u5jFevsIQAB4ADqJQuDDUIAhgwOLIGVLUCADWADwA0sjpkCCkWMiVEACeojDI6MboAHwAFBmUtcZBYJRQEHCk4lIdveoAlJQAglBQcB3V6Cq1agN+6Vk5yHkFRSVlFTUACg1pTS3I3HzDo8j3a8h7w5MNlsdtUAEoQMRQUjVe79AYDFZaAbLFKnbK5fKFYogUrlKpDAB0RLgUF4WEocBAHVWFKp+FYMzA1CgOPUfiEBEuWJKkBw-yGkwUKImsjAIRRv32GkIxEZzNZKQ5YgkuBmrW0KiJBNu+LMZDMIuCKxSyskvEivLAQxURmQDjUxk1RLVahWflNuF4cgtymttvtjq1OsqQz1pAN0lFDmNbqAA

**Related Issues:**

**Comment by sooth-sayer** [comment]
It works as expected if I remove 
`export function pluck<P extends string>(p: P): <T>(list: Array<Record<P, T>>) => T[];`
overload

**Comment by uhyo** [comment]
This bug seems to be fixed in TypeScript 4.0. :tada:

**Comment by RyanCavanaugh** [comment]
Thanks for pointing it out @uhyo 🙂
